rust/agents/validator/src/submit.rs

```diff
@@ -133,8 +142,20 @@ impl ValidatorSubmitter {
                 continue;
             }
 
-            self.submit_checkpoints_until_correctness_checkpoint(&mut tree, &latest_checkpoint)
-                .await?;
+            tree = call_and_retry_indefinitely(|| {
```

Comment:
Need to explicitly return `tree` and reassign to keep `tree` updated, because the async closure needs to take ownership instead of mutably borrowing